src/compiler/checker.ts

```diff
@@ -865,6 +865,11 @@ namespace ts {
                                 error(errorLocation, Diagnostics.Static_members_cannot_reference_class_type_parameters);
                                 return undefined;
                             }
+                            // Only perform additional check if error reporting was requested
+                            if (nameNotFoundMessage && !isTypeParameterSymbolDeclaredInContainer(result, location)) {
+                                error(errorLocation, Diagnostics.Type_parameter_0_cannot_be_referenced_outside_of_the_declaration_that_defines_it, symbolToString(result));

```

Comment:
In the design meeting we discussed this being an error, however I can see the benefits of just silently skipping the type parameter as well.